ArduCopter/mode_guided.cpp

```diff
@@ -286,47 +321,86 @@ bool ModeGuided::set_destination(const Location& dest_loc, bool use_yaw, float y
         pos_control_start();
     }
 
-    if (!wp_nav->set_wp_destination_loc(dest_loc)) {
-        // failure to set destination can only be because of missing terrain data
-        AP::logger().Write_Error(LogErrorSubsystem::NAVIGATION, LogErrorCode::FAILED_TO_SET_DESTINATION);
-        // failure is propagated to GCS with NAK
-        return false;
-    }
-
     // set yaw state
     set_yaw_state(use_yaw, yaw_cd, use_yaw_rate, yaw_rate_cds, relative_yaw);
 
+    // set position target and zero velocity and acceleration
+    if (!dest_loc.get_vector_from_origin_NEU(guided_pos_target_cm)) {
+        guided_pos_target_cm = pos_control->get_pos_target_cm();
+    }
+    guided_vel_target_cms.zero();
+    guided_accel_target_cmss.zero();
+    vel_update_time_ms = millis();
+
     // log target
-    copter.Log_Write_GuidedTarget(guided_mode, Vector3f(dest_loc.lat, dest_loc.lng, dest_loc.alt),Vector3f());
+    copter.Log_Write_GuidedTarget(guided_mode, Vector3f(dest_loc.lat, dest_loc.lng, dest_loc.alt), guided_vel_target_cms, guided_accel_target_cmss);
 
     send_notification = true;
 
     return true;
 }
 
-// guided_set_velocity - sets guided mode's target velocity
+// set_velaccel - sets guided mode's target velocity and acceleration
+void ModeGuided::set_accel(const Vector3f& acceleration, bool use_yaw, float yaw_cd, bool use_yaw_rate, float yaw_rate_cds, bool relative_yaw, bool log_request)
+{
+    // check we are in velocity control mode
+    if (guided_mode != SubMode::Accel) {
+        accel_control_start();
+    }
+
+    // set yaw state
+    set_yaw_state(use_yaw, yaw_cd, use_yaw_rate, yaw_rate_cds, relative_yaw);
+
+    // set velocity and acceleration targets and zero position
+    guided_pos_target_cm.zero();
+    guided_vel_target_cms.zero();
+    guided_accel_target_cmss = acceleration;
+    vel_update_time_ms = millis();
```

Comment:
we seem to have two timers (posvelaccel_update_time_ms and vel_update_time_ms) but I think we should have either one or three.

If we have only one, it should probably be called, "posvelaccel_update_time_ms".  If we have three they should probably be "posvelaccel_update_time_ms", "velaccel_update_time_ms" and "accel_update_time_ms".

Reply:
update_time_ms only